### ISSUE TITLE
mask manager is fully broken

### ISSUE THREAD

**Issue by TurboGit** [darktable-org/darktable#5359]
**Describe the bug**
Shapes are not visible when creating them or selecting them.
It is not possible to create a brush or a path.

**To Reproduce**
1. Go to darkroom
2. Open mask manager
3. Click on circle
4. Move over image, no shape displayed
5. Click to place it, entry added in mask manager
6. Select it, no mask displayed on the image
7. Select path or brush - cannot create it

**Expected behavior**
It should be possible to see and create all kind of shapes.

**Platform (please complete the following information):**
 - Darktable Version: master
 - OS: GNU/Debian

**Additional context**

**Comment by aurelienpierre** [comment]
It works on b05487b63b14a562078f6ec7d7a316c76eb0c7a4, so issue was introduced later.

**Comment by aurelienpierre** [comment]
If you open a module while any drawing tool is enabled in mask manager, that makes it appear. So it looks like a `gui_expose()` issue somewhere.

**Comment by aurelienpierre** [comment]
My bad, the issue is much older, it went unnoticed because, as long as a module is expanded, everything displays fine. So there is some conditional `gui_expose()` triggered conditionaly if a module has focus, that should be triggered no matter what.

**Comment by aurelienpierre** [comment]
more news, when adding a new mask in mask manager, it needs a module as a root:
`_tree_add_circle(GtkButton *button, dt_iop_module_t *module)`.

It seems that, if no module is active/in-focus, module is NULL and mask creation is bypassed.

**Comment by TurboGit** [comment]
Probably so. And the module does not have only to be expanded but also active.

**Comment by aurelienpierre** [comment]
In that case, I guess `dt_iop_module_t *module` should default to `mask_manager.c` if no module is active. But that sounds like this error has been there for a long time.

**Comment by TurboGit** [comment]
Right !

**Comment by aurelienpierre** [comment]
That should be in `src/libs/masks.c:941` :

```
  dt_iop_module_t *module = NULL;
  int on_row = 0;
  if(gtk_tree_view_get_path_at_pos(GTK_TREE_VIEW(treeview), (gint)event->x, (gint)event->y, &mouse_path, NULL,
                                   NULL, NULL))
  {
    on_row = 1;
    // we retrieve the iter and module from path
    if(gtk_tree_model_get_iter(model, &iter, mouse_path))
    {
      GValue gv = {
        0,
      };
      gtk_tree_model_get_value(model, &iter, TREE_MODULE, &gv);
      module = g_value_peek_pointer(&gv);
      g_value_unset(&gv);
    }
  }
```
But I don't get what the Gtk pointer abstraction (treeview -> g_value_peek_pointer) is doing. There is no explicit iop list in there, so I don't see how to address directly the mask manager iop to init `dt_iop_module_t *module`

**Comment by TurboGit** [comment]
I think this code is there since long time. It get the module pointer out of the tree (this is to handle masks created from a specific module).

**Comment by aurelienpierre** [comment]
How do you target a specific element in the tree ? Fetch `dev->iop_order_list`, get the position of `mask_manager`, and hope it has the same position in the treeview ?

**Comment by TurboGit** [comment]
This is the culprit : 5b0289627fa1e31df251c25a0d1177075c7acf29

commit 5b0289627fa1e31df251c25a0d1177075c7acf29
Author: hanno@schwalm-bremen.de <hanno@schwalm-bremen.de>
Date:   Sat Mar 14 08:55:30 2020 +0100

    Only show forms and color picker when module is enabled
    
    The colorpicker and mask forms visualizers for a module should only
    be drawn in darkroom if the module is currently enabled.
    
    Fixes several issues #3532, #3514, last part of #3451, #3642

 src/views/darkroom.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

**Comment by TurboGit** [comment]
Now fixing...

**Comment by aurelienpierre** [comment]
Oooh, nice catch.

**Comment by TurboGit** [comment]
Fixed by b1d8484da6cca3b90a4e08c003b6fc62351e004d.
